### PR TITLE
Linux support, dep cleanup

### PR DESCRIPTION
**Pull request by Jakobzs**

### CODE REVIEW

File: rapl-lib/src/rapl/linux.rs [hunk 1]
```diff
@@ -1,75 +1,127 @@
...
+    os::unix::prelude::FileExt,
+    sync::Once,
+};
 
-// Impl this:
-// https://github.com/greensoftwarelab/Energy-Languages/blob/master/RAPL/rapl.c#L14
+// Running it for now: sudo ./target/debug/rapl-bin
+
+const AMD_MSR_PWR_UNIT: i64 = 0xC0010299;
+const MSR_RAPL_POWER_UNIT: i64 = 0x606;
```
Comment (sabj17) [inline rapl-lib/src/rapl/linux.rs:12]: change to INTEL_MSR_PWR_UNIT

File: rapl-lib/src/rapl/linux.rs [hunk 2]
```diff
@@ -1,75 +1,127 @@
...
-        let pread_err = CString::new("rdmsr:pread").unwrap();
-        unsafe { perror(pread_err.as_ptr()) };
-    }
+    let mut output_data: [u8; 8] = [0; 8];
 
-    //println!("val: {}", val);
+    // TODO: Consider just seek here instead, same impl for Windows then
+    f.read_at(&mut output_data, msr_offset as u64).unwrap();
+    //let num_bytes_read = fd.read_at(&mut output_data, msr_offset as u64).unwrap();
+    //println!("number of bytes read: {}", num_bytes_read);
```
Comment (Villiamfj) [inline rapl-lib/src/rapl/linux.rs]: hmm udkommenteret kode 🤔

File: rapl-lib/src/rapl/linux.rs [hunk 3]
```diff
@@ -1,75 +1,127 @@
...
+    }
 }
 
 // https://github.com/greensoftwarelab/Energy-Languages/blob/master/RAPL/rapl.c#L157
-fn rapl_init() {}
+// fn rapl_init() {} // Use OnceCell for init e.g. with fd
 
 // https://github.com/greensoftwarelab/Energy-Languages/blob/master/RAPL/rapl.c#L64
-fn detect_cpu() {}
+// fn detect_cpu() {} // Compile timed currently
```
Comment (Villiamfj) [inline rapl-lib/src/rapl/linux.rs]: flyt comment

**Review by Villiamfj — APPROVED** [review]

**Review by sabj17 — APPROVED** [review]
Do better

**Pull request merged**